server/src/main/java/org/elasticsearch/index/analysis/Analysis.java

```diff
@@ -303,19 +304,19 @@ public static Reader getReaderFromIndex(
         SynonymsManagementAPIService synonymsManagementAPIService
     ) {
         // TODO: this is a temporary solution for loading synonyms under feature flag, to be redesigned for GA
-        final PlainActionFuture<SynonymsManagementAPIService.SynonymsSetResult> synonymsLoadingFuture = new PlainActionFuture<>() {
```

Comment:
FYI @mayya-sharipova, some changes to the Analyzer loading code due to the change in results for `getSynonymRules`